docs/ocp4-manual-deployment.md

```diff
@@ -1,567 +1,209 @@
-
-**5. Install and setup OC client**
-
-This can be done on any host you prefer to use for interacting with the cluster. In this document `oc` is setting up the bastion host itself
-
-a. Download `https://mirror.openshift.com/pub/openshift-v4/ppc64le/clients/ocp/candidate-4.3/openshift-client-linux-4.3.18.tar.gz` and extract and make the oc binary available in PATH.
-
-b. Create /root/.kubeconfig file
-   The file is present in the OpenShift installer’s auth directory.
-
-**6. Approve all pending CSR requests from the bootstrap MCO and the worker nodes**
-
-Run `oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs oc adm certificate approve`
-
-**7. Wait for all cluster operators to come online**
-Run  `oc get co`
-
-The authentication, ingress and monitoring co's will take some time to be Available. Review for any pending CSR’s in the queue and approve them again. Wait for some more time for the pending co's to come online.
-
-**8.  Patch the image config registry**
-
-The image-registry is not always available immediately after the OCPinstaller
+1. Bootstrap
+2. Masters
+3. Workers
 
-Run below
+## **Wait for Install**
 
 ```
-while [ $(oc get configs.imageregistry.operator.openshift.io/cluster | wc -l) == 0 ]; do sleep 30; done
-oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}, "managementState": "Managed"}}'
+openshift-install wait-for bootstrap-complete --log-level debug
 ```
-**9.  Complete the installation**
 
-Once all the cluster Operators are AVAILABLE, run
+## **Finish Install
+First, login to your cluster
+```
+export KUBECONFIG=/root/ocp4/auth/kubeconfig
+```
+Set the registry for your cluster
 
-`openshift-install --dir=<installation_directory> wait-for install-complete`
+First, you have to set the `managementState` to `Managed` for your cluster
+```
+oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState":"Managed"}}'
+```
+For PoCs, using emptyDir is ok (to use PVs follow this doc)
+```
+oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}}}'
+```
+If you need to expose the registry, run this command
+```
+oc patch configs.imageregistry.operator.openshift.io/cluster --type merge -p '{"spec":{"defaultRoute":true}}'
+```
+**Note** - You can watch the operators running with ```oc get clusteroperators```
 
+Watch your CSRs. These can take some time; go get come coffee or grab some lunch. You'll see your nodes' CSRs in "Pending" (unless they were "auto approved", if so, you can jump to the wait-for install-complete step)
+```
+watch oc get csr
+```
+To approve them all in one shot...
+```
+oc get csr --no-headers | awk '{print $1}' | xargs oc adm certificate approve
 ```
-INFO Waiting up to 30m0s for the cluster to initialize...
+Check for the approval status (it should say "Approved,Issued")
```

Comment:
CSRs are created not in any order or all at once. We should mention that approve all pending CSR until all the worker nodes are in "Ready" status.

Reply:
true.. thanks for pointing it out